src/decawave/src/decawave_node.cpp

```diff
@@ -110,29 +85,28 @@ void update_callback(const ros::TimerEvent&){
   decawave::Range msg;
 
   // update decawave data
-  piTag->updateSamples();
-
-  // get tag position from the decawave
-  tagPos = piTag->getPos();
-
-  ros::param::get("~port_num",port_num);
-  std::string frame_id = std::to_string(port_num);
-
-  msg.distance = tagPos.x;
-  msg.estimated_variance = 0.10;
-  // fill out message header
-  msg.header.stamp = ros::Time::now();
-  msg.header.frame_id = "decawave" + frame_id;//"decawave_" + port_num;
-  msg.child_frame_id = "decawave2_link"; //change to correct part
-
-  gps_pub.publish(msg);
-
-  msg.distance = tagPos.y;
-  msg.estimated_variance = 0.10;
-  // fill out message header
-  msg.header.stamp = ros::Time::now();
-  msg.header.frame_id = "decawave" + frame_id;//"decawave_" + port_num;
-  msg.child_frame_id = "decawave3_link"; //change to correct part
-
-  gps_pub.publish(msg);
+  std::vector<anchor> anchors = piTag->updateSamples();
+
+  ros::param::get("~port_num",port_name);
+  std::string frame_id = port_name;
+
+  int j=anchors.size();
+  int i=0;
+  anchor m_anchor;
+  while (i<j){
```

Comment:
Ok, this is also a for loop. We can do this better:
The pre-C++11 way to iterate through a vector (especially if you just want a constant read-only reference to the data!) is with iterators:
http://www.cplusplus.com/reference/vector/vector/cbegin/
Now we can write, using const for read-only: 
```
for (const Anchor &anchor : achors) {
```
https://stackoverflow.com/questions/15027282/c-for-each-pulling-from-vector-elements